src/formBuilder/utils.js

```diff
@@ -801,13 +802,7 @@ export function generateUiSchemaFromElementProps(
 
   elementArr.forEach((element) => {
     uiOrder.push(element.name);
-    if (element.$ref !== undefined) {
-      // look for the reference
-      const pathArr =
-        typeof element.$ref === 'string' ? element.$ref.split('/') : [];
-      if (definitions && definitions[pathArr[2]])
-        uiSchema[element.name] = definitions[pathArr[2]];
```

Comment:
I'm not sure we can just remove this.  We did remove it just to test the idea, but I think we still need to keep it (I am not sure what implications removing this will have).

Can we instead extract out where the uiOptions are fetched (lines 811 onwards) and use that code in both ref and non-ref cases?